### PR TITLE
Events on calendar

### PR DESCRIPTION
**Pull request by pandafulmanda**
- [x] events adding on calendar page
- [x] update add menu tests to include external and events
- [x] added route and placeholder shell for event
- [x] clean up event edit/create page to match [mockup](http://mockups.openstax.org/Tutor_MVP_future_release/start.html#p=add_event_-_teacher)
  * might be further updates after UX discussion
- [x] test with the BE at some point
  * openstax/tutor-server#683
  * openstax/tutor-server#686
- [x] show for student
  - needs icon

screen shot 2015-09-29 at 11 46 13 am
screen shot 2015-09-29 at 11 46 41 am
screen shot 2015-09-29 at 12 46 14 pm
screen shot 2015-09-30 at 3 13 42 pm
screen shot 2015-09-30 at 3 35 12 pm

### CODE REVIEW

**Comment by jpslav** [comment]
@pandafulmanda I just moved Joe's openstax/tutor-server#592 over to openstax/tutor-server#683 and merged it to master.  Let me and @joemsak know if there are any problems.

File: src/components/course-calendar/plan-event-details.cjsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  render: ->
+    {plan, courseId, className} = @props
+    {title, type, id} = plan
+    linkParams = {courseId, id}
+    editLinkName = camelCase("edit-#{type}")
+
+    viewOrEdit = if plan.isEditable then 'Edit' else 'View'
+    editButton = <Router.Link to={editLinkName} params={linkParams}>
+      <BS.Button className='-edit-assignment'>{viewOrEdit} Event</BS.Button>
```
Comment (philschatz) [inline src/components/course-calendar/plan-event-details.cjsx]: Could this be just a link with a className='btn btn-default' ? seems odd to have a `<button>` inside an `<a>`

File: src/components/course-calendar/plan.cjsx [hunk 2]
```diff
@@ -185,30 +201,35 @@ CoursePlan = React.createClass
...
-          plan={plan}
-          courseId={courseId}
-          className={planClasses}
-          onRequestHide={@syncIsViewingStats.bind(null, false)}
-          ref='details'/>
+      modalProps =
+        plan: plan
+        courseId: courseId
+        className: planClasses
+        onRequestHide: @syncIsViewingStats.bind(null, false)
```
Comment (philschatz) [inline src/components/course-calendar/plan.cjsx]: Why not `_.partial()`?

File: src/components/markdown.cjsx [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 Markdown = require 'markdown-it'
 ArbitraryHtml = require './html'
+_ = require 'underscore'
 
 md = new Markdown()
 
 module.exports = React.createClass
   render: ->
     {text} = @props
+    htmlProps = _.pick(@props, 'block', 'className')
```
Comment (philschatz) [inline src/components/markdown.cjsx]: Would it be easier to `_.omit(@props, 'text')`?

**Pull request merged**